File: java/src/min_task_time/MinTaskTimeTest.java
package min_task_time;

import org.junit.Assert;
import org.junit.Test;

import java.util.Arrays;

public class MinTaskTimeTest {
	@Test
	public void testExample() {
		Assert.assertEquals(120, MinTaskTime.minTaskTime(5,
			new int[]{20, 10, 30, 50, 40},
			new int[][]{{3, 4}, {2, 3}, {1, 0}, {2, 1}, {3, 1}, {3, 0}}));
	}

	@Test
	public void testRandomCases() {
		for (int test = 0; test < 10_000; test++) {
			var td = new randomTestData();
			var actual = MinTaskTime.minTaskTime(td.n, td.time, td.before);
			if (td.expected != actual) {
				Assert.fail(String.format("minTaskTime(%s, %s, %s):\nwant %s\n got %s",
					td.n, Arrays.toString(td.time), Arrays.deepToString(td.before), td.expected, actual));
			}
		}
	}
}

File: python/min_task_time/min_task_time_test.py
from min_task_time import min_task_time
from mtt_test_data import RandomTestData


def test_example():
	got = min_task_time(5, [20, 10, 30, 50, 40], [[3, 4], [2, 3], [1, 0], [2, 1], [3, 1], [3, 0]])
	assert got == 120


def test_random_cases():
	for _ in range(5_000):
		td = RandomTestData()
		got = min_task_time(td.n, td.time, td.before)

		assert got == td.want, \
			f'min_task_time({td.n}, {td.time}, {td.before}):\nwant {td.want}\n got {got}'
